### ISSUE TITLE
String truncation (shortening)

### ISSUE THREAD

**Issue by 5H4D0W-X** [Yellow-Dog-Man/Resonite-Issues#2037]
### Is your feature request related to a problem? Please describe.

It can be very useful to take an arbitrary string and truncate it so that it can be displayed correctly in UI without scaling it.

### Describe the solution you'd like

- An option on TextRenderer and Text to truncate text to fit its bounds with an optional suffix (such as "...")
- A flux node that truncates an input string based on a multiple of its height (i.e. a truncation with factor 5 would result in a string about 5 times as long as it's high), also with an optional suffix.

### Describe alternatives you've considered

Using a hidden text renderer and iteratively measuring its size while removing characters until a target size is reached. This is hard to implement, especially dynamically where the size of UI elements may change at any time.

### Additional Context

This should be able to deal with resized rich text as well, that should be taken into account.

### Requesters

ShadowX

**Comment by epicEaston197** [comment]
I'm having trouble understanding this 
do you mean something like this?
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/76523692/1b7b8330-10e3-4d86-a1e3-c12dbe8be733)

**Comment by 5H4D0W-X** [comment]
Exactly, it would make things like world names in the contacts list much more consistent
